radixdlt/src/main/java/com/radixdlt/consensus/VertexStore.java

```diff
@@ -69,10 +70,14 @@ public VertexStore(
 		this.engine = Objects.requireNonNull(engine);
 		this.counters = Objects.requireNonNull(counters);
 		this.highestQC = Objects.requireNonNull(rootQC);
-		try {
-			this.engine.checkAndStore(genesisVertex.getAtom());
-		} catch (RadixEngineException e) {
-			throw new IllegalStateException("Could not store genesis atom: " + genesisVertex.getAtom(), e);
+
+		if (genesisVertex.getAtom() != null) {
+			CommittedAtom committedGenesis = genesisVertex.getAtom().committed(rootQC.getProposed());
+			try {
+				this.engine.checkAndStore(committedGenesis);
```

Comment:
Maybe now would be a good time to only store if not already present?

Reply:
I think this can be saved for later